lib/application.js

```diff
@@ -607,7 +606,7 @@ app.render = function render(name, options, callback) {
  */
 
 app.listen = function listen () {
-  var server = http.createServer(this)
+  var server = this.createServer(this)
```

Comment:
Honestly I think this should just deprecate this in favor of people explicitly writing `http.createServer(app).listen(...)`. It also removes one more option from the `createApplication` options which is only ever used once here. It took me a while to follow the indirect assignments around and I think that's bad for developers and users to understanding what's going on.

But I do understand why it's done after 30 minutes of review and won't block on this issue. I would be in favor of refactoring this to assign `app.listen` within the create function instead. That way there's less indirect dependencies that's hard to reason about and `createServer` is used right where it's defined.

Reply:
> I would be in favor of refactoring this to assign app.listen within the create function instead.

Of the options to achieve this I think this one is my preference. I believe any of these loose the `once` behavior, but I am not sure it makes a big difference but I agree simplifying things like this is good. I honestly cannot remember if there used to be cases where the server could emit `error` events more than once, but maybe that was an old node behavior?